### PR TITLE
Salt front munge seed

### PR DESCRIPTION
**Pull request by galengorski**
This PR just has a few changes:

- I added a script for cleaning up the salt front record and saving it in `03a_it_analysis/in`
- I also update the set seed function in the `run_model.py` script, this way each replicate has a different seed. Now running 5 replicates will produce 5 different results, but running those same replicates again will produce the same 5 results.

### CODE REVIEW

File: 02_munge/src/salt_front_record.R [hunk 1]
```diff
@@ -0,0 +1,25 @@
+#===================================================================================#
```
Comment (amsnyder) [inline 02_munge/src/salt_front_record.R:1]: Should this file get added into the Snakefile for fetch/munge?

File: 03b_model/src/run_model.py [hunk 2]
```diff
@@ -557,10 +561,10 @@ def make_predictions(prepped_model_io_data_file, target,
     #unnormalize
     #predictions for train val set
     preds_trainval_c  = preds_trainval.detach().numpy().reshape(preds_trainval.shape[0]*preds_trainval.shape[1],preds_trainval.shape[2])
-    unnorm_trainval = ((preds_trainval_c*means_stds['y_std_trnval'][target].data)+means_stds['y_mean_trnval'][target].data).squeeze()
+    unnorm_trainval = ((preds_trainval_c*means_stds['y_std_trn'][target].data)+means_stds['y_mean_trn'][target].data).squeeze()
```
Comment (amsnyder) [inline 03b_model/src/run_model.py:564]: I'm not sure what this switch was - is this something that should be set in the config file if it will be toggled?

File: 03b_model/src/run_model.py [hunk 3]
```diff
@@ -724,21 +729,21 @@ def test_hyperparameters():
             print('Running hyperparameter-replicate combination '+str((j+1)*(i+1))+' of '+str(len(hp_tune_vals)*hp_config['replicates']))
 
             run_id = os.path.join(hp_id, str(i).rjust(2,'0'))
-            
+            seed = i
             train_model(prepped_model_io_data_file, inputs, seq_len,
```
Comment (amsnyder) [inline 03b_model/src/run_model.py:733]: You have some commented out code that uses the train_model() and make_predictions() functions. Should those be updated to get the new inputs too, if there a possibility that you would uncomment them at some point?

File: 03b_model/src/run_model.py [hunk 4]
```diff
@@ -630,20 +634,21 @@ def run_replicates(n_reps, prepped_model_io_data_file):
         
         run_id = os.path.join(config['run_id'], str(i).rjust(2,'0'))
         
+        seed = i
```
Comment (amsnyder) [inline 03b_model/src/run_model.py:637]: I'm not sure I understand what's going on with setting the see in the config file and then resetting it based on the iteration number here. Is this a different seed? and if so, maybe it should get a different name. I'm not sure what happens when you have a global variable name 'seed' and you create a local variable with the same name inside a function. Either way, it would be good to give them different names for clarity.

**Review by amsnyder — CHANGES_REQUESTED** [review]
just a few small comments!

File: 03b_model/src/run_model.py (hunk 3 above)
Comment (galengorski) [reply 03b_model/src/run_model.py:733]: Removed the commented code, it was old and we won't be using it.

File: 03b_model/src/run_model.py (hunk 2 above)
Comment (galengorski) [reply 03b_model/src/run_model.py:564]: This is a small change to how the data are normalized and unnormalized before and after training, respectively. They were initially normalized/unnormalized with the training and validation data, but now it is just with the training data. It shouldn't have a big effect and I don't think we'll need to change/toggle it.

File: 03b_model/hyperparameter_config.yaml [hunk 5]
```diff
@@ -1,4 +1,4 @@
-out_dir: '03b_model/out/HP_Tune_01'
+out_dir: '../../../../../03b_model/out/HP_Tune_02'
```
[the remark this answers is not in the document]
Comment (galengorski) [reply 03b_model/hyperparameter_config.yaml]: Changed to relative path

File: 03b_model/src/run_model.py (hunk 4 above)
Comment (galengorski) [reply 03b_model/src/run_model.py:637]: I added some explanatory text in the `model_config.yaml` file, basically, the idea is that the `seed_set` parameter in the config file only pertains to a single model run using the `main()` function. For the `run_replicates()` and `test_hyperparamters()` functions, the seed is set within those functions as the iterator. This way the seed is unique between replicates, but reproducible.

File: 02_munge/src/salt_front_record.R (hunk 1 above)
Comment (galengorski) [reply 02_munge/src/salt_front_record.R:1]: Yes, it should, I'll open an issue for that.

File: 03b_model/src/run_model.py (hunk 4 above)
Comment (amsnyder) [reply 03b_model/src/run_model.py:637]: Do you think the seed used in the iterations could be named something like `replicate_seed`? The global variable `seed` is visible inside and outside of functions. Then the local variable `seed` is only visible inside the function it is created within. So inside that function, there is some ambiguity about if you are using the global `seed` or the local `seed`. I imagine it is using the local variable, but giving these variables different names would help clear up any ambiguity.

Comment (galengorski) [reply 03b_model/src/run_model.py]: Good idea! Changes made

**Review by amsnyder — APPROVED** [review]
